src/wp-includes/nav-menu.php

```diff
@@ -780,6 +759,41 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 	return apply_filters( 'wp_get_nav_menu_items', $items, $menu, $args );
 }
 
+/**
+ * Prime all linked objects to menu items.
+ *
+ * @param WP_Post[] $menu_items Array post objects of menu items.
+ */
+function _prime_menu_items_objects( $menu_items ) {
+	$post_ids = array();
+	$term_ids = array();
+
+	foreach ( $menu_items as $menu_item ) {
+		if ( 'nav_menu_item' !== $menu_item->post_type ) {
+			continue;
+		}
+		$object_id = get_post_meta( $menu_item->ID, '_menu_item_object_id', true );
+		$type      = get_post_meta( $menu_item->ID, '_menu_item_type', true );
+
+		if ( 'post_type' === $type ) {
+			$post_ids[] = (int) $object_id;
+		} elseif ( 'taxonomy' === $type ) {
+			$term_ids[] = (int) $object_id;
+		}
+	}
+
+	if ( ! empty( $post_ids ) ) {
+		_prime_post_caches( $post_ids, false );
+	}
+
+	unset( $post_ids );
+
+	if ( ! empty( $term_ids ) ) {
+		_prime_term_caches( $term_ids );
+	}
+	unset( $term_ids );
```

Comment:
I don't think there is any need for these to be `unset` here. They'll be cleaned up by PHP when the function exits.

Reply:
This was just existing logic I copied over. Should I remove it?

Reply:
Yeah I think it was probably in the original function to clean up scope.